Source/DafnyCore/Backends/Dafny/ASTBuilder.cs

```diff
@@ -2,12 +2,18 @@
 using System.Collections.Generic;
 using Dafny;
 using DAST;
+using DAST.Format;
+using JetBrains.Annotations;
 using Microsoft.CodeAnalysis;
 using Microsoft.Dafny.Compilers;
 using Std.Wrappers;
 
 namespace Microsoft.Dafny.Compilers {
 
+  interface UnsupportedContainer {
```

Comment:
Naming threw me off, I think you just want "Container", and independently all containers must support `AddUnsupported`